src/core/weightedGraph2.js

```diff
@@ -37,89 +65,252 @@ export type WeightedEdge = {|
  * WeightedGraph2 should completely replace it.
  */
 export class WeightedGraph2 {
-  _nodeWeights: NodeTrie<NodeWeight>;
-  _edgeWeights: EdgeTrie<EdgeWeight>;
-  graph: Graph;
+  _weights: WeightsI;
+  _graph: Graph;
```

Comment:
My sense is that these should be immutable.  Another advantage of using an object is that it would properly shield this data in its closure.

Reply:
@elihanover these can't be immutable because WG2 operates on them. WG2 is meant to be make it easy to build a WeightedGraph and not just read a WeightedGraph.

Reply:
One thing I'm not sure about: Do we want to utilize a `copy` of each of these instead of the original reference?

Reply:
That's how I did `Weights`.  You know you're working with a copy each time, so you don't run into any issues of handling/sharing state in multiple places.  Seems more pure.

Reply:
That's how I did `Weights`.  You know you're working with a copy each time, so you don't run into any issues of handling/sharing state in multiple places.  Seems more pure.